src/main/java/org/betonquest/betonquest/objectives/BlockObjective.java

```diff
@@ -78,4 +106,31 @@ public void stop() {
         HandlerList.unregisterAll(this);
     }
 
+    private boolean checkLocation(final Location loc, final Profile profile) {
+        try {
+            if (location != null) {
+                if (location2 != null) {
+                    return isInRange(loc, profile);
+                }
+                return loc.equals(location.getLocation(profile));
```

Comment:
I think block locations are always at .5 while in BetonQuest you usually enter integers (or have a imprecise location in a variable). So we need to compare block coordinates and not exact coordinates.